### PR TITLE
Company admin

### LINKED ISSUE #7: Admin can manage companies
Once the company is registered (#3) the admin should verify their account.

### PR DESCRIPTION
**Pull request by rtio**
it closes #7

### CODE REVIEW

File: src/Entity/Company.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+namespace App\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\CompanyRepository")
+ */
+class Company
```
Comment (ianrodrigues) [inline src/Entity/Company.php]:
```suggestion
final class Company
```

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,29 @@
 deploy:
```
Comment (ianrodrigues) [inline Makefile]: Out of scope of this PR.

File: src/Entity/Company.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        return $this;
+    }
+
+    public function getLogo(): ?string
+    {
+        return $this->logo;
+    }
+
+    public function setLogo(?string $logo): self
```
Comment (ianrodrigues) [inline src/Entity/Company.php]:
```suggestion
    public function setLogo(string $logo): self
```

File: src/Entity/Company.php [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+        return $this;
+    }
+
+    public function getAddress(): ?string
+    {
+        return $this->address;
+    }
+
+    public function setAddress(?string $address): self
```
Comment (ianrodrigues) [inline src/Entity/Company.php]:
```suggestion
    public function setAddress(string $address): self
```

File: src/Entity/Company.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+     */
+    private $address;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private $email;
+
+    /**
+     * @ORM\Column(type="string", length=255, nullable=true)
```
Comment (ianrodrigues) [inline src/Entity/Company.php]: Shouldn't it be required?

File: src/Entity/Company.php [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+        return $this;
+    }
+
+    public function getPassword(): ?string
+    {
+        return $this->password;
+    }
+
+    public function setPassword(?string $password): self
```
Comment (ianrodrigues) [inline src/Entity/Company.php]:
```suggestion
    public function setPassword(string $password): self
```

File: src/Repository/CompanyRepository.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+use App\Entity\Company;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+/**
+ * @method Company|null find($id, $lockMode = null, $lockVersion = null)
+ * @method Company|null findOneBy(array $criteria, array $orderBy = null)
+ * @method Company[]    findAll()
+ * @method Company[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
```
Comment (ianrodrigues) [inline src/Repository/CompanyRepository.php]: I don't think we should maintain this kind of doc block. Pretty decent IDEs can handle with the class' methods.

File: src/Entity/Company.php [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+            // set the owning side to null (unless already changed)
+            if ($jobOffer->getCompany() === $this) {
+                $jobOffer->setCompany(null);
+            }
+        }
+
+        return $this;
+    }
+
+    public function __toString(): string
```
Comment (ianrodrigues) [inline src/Entity/Company.php]: It'd be great to have unit tests for that.

File: src/Entity/JobOffer.php [hunk 9]
```diff
@@ -127,4 +128,9 @@ public function setStatus(string $status): self
 
         return $this;
     }
+
+    public function __toString(): string
```
Comment (ianrodrigues) [inline src/Entity/JobOffer.php]: Unit tests for that, also.

File: src/Entity/Company.php (hunk 5 above)
Comment (rtio) [reply src/Entity/Company.php]: it isn't required but keeps explicit.

File: src/Repository/CompanyRepository.php (hunk 7 above)
Comment (rtio) [reply src/Repository/CompanyRepository.php]: done

File: src/Entity/Company.php (hunk 8 above)
Comment (rtio) [reply src/Entity/Company.php]: done

File: src/Entity/JobOffer.php (hunk 9 above)
Comment (rtio) [reply src/Entity/JobOffer.php]: done

File: factories/JobOfferFactory.php [hunk 10]
```diff
@@ -8,7 +8,7 @@
 $fm->define(JobOffer::class)->setDefinitions([
     'title' => Faker::jobTitle(),
     'description' => Faker::text(),
-    'company' => Faker::company(),
+    'company' => "entity|App\Entity\Company",
```
Comment (ianrodrigues) [inline factories/JobOfferFactory.php]:
```suggestion
    'company' => 'entity|App\Entity\Company',
```

File: tests/Feature/DisplaySingleJobOfferTest.php [hunk 11]
```diff
@@ -6,16 +6,31 @@
...
+use App\Entity\Company;
 use Symfony\Component\HttpFoundation\Response;
 
 class DisplaySingleJobOfferTest extends TestCase
 {
+    private $company;
+
+    public function setUp(): void
+    {
+        parent::setUp();
```
Comment (ianrodrigues) [inline tests/Feature/DisplaySingleJobOfferTest.php]:
```suggestion
        parent::setUp();

```

File: tests/Feature/ListJobOffersTest.php [hunk 12]
```diff
@@ -4,28 +4,38 @@
...
 use App\Entity\JobOffer;
 use Symfony\Component\HttpFoundation\Response;
 
 class ListJobOffersTest extends TestCase
 {
+    private $company;
+
+    public function setUp(): void
+    {
+        parent::setUp();
```
Comment (ianrodrigues) [inline tests/Feature/ListJobOffersTest.php]:
```suggestion
        parent::setUp();

```

File: tests/Unit/Entity/CompanyTest.php [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+class CompanyTest extends TestCase
+{
+
+    public function test__toString()
+    {
+        $company = $this->factory->create(Company::class, [
+            'name' => 'Dunder Mifflin',
+        ]);
+        $this->assertEquals("#{$company->getId()} Dunder Mifflin", (string) $company);
```
Comment (ianrodrigues) [inline tests/Unit/Entity/CompanyTest.php]:
```suggestion

        $this->assertEquals("#{$company->getId()} Dunder Mifflin", (string) $company);
```

File: tests/Unit/Entity/CompanyTest.php [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+declare(strict_types=1);
+
+namespace App\Tests\Unit\Entity;
+
+use App\Entity\Company;
+use App\Tests\TestCase;
+
+class CompanyTest extends TestCase
+{
+
```
Comment (ianrodrigues) [inline tests/Unit/Entity/CompanyTest.php]:
```suggestion
```

File: tests/Unit/Entity/CompanyTest.php [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+    public function test__toString()
```
Comment (ianrodrigues) [inline tests/Unit/Entity/CompanyTest.php]:
```suggestion
    public function test_string_representation()
```

File: tests/Unit/Entity/JobOfferTest.php [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+declare(strict_types=1);
+
+namespace App\Tests\Unit\Entity;
+
+use App\Entity\JobOffer;
+use App\Tests\TestCase;
+
+class JobOfferTest extends TestCase
+{
+
```
Comment (ianrodrigues) [inline tests/Unit/Entity/JobOfferTest.php]:
```suggestion
```

File: tests/Unit/Entity/JobOfferTest.php [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+    public function test__toString()
```
Comment (ianrodrigues) [inline tests/Unit/Entity/JobOfferTest.php]:
```suggestion
    public function test_string_representation()
```

File: tests/Unit/Entity/JobOfferTest.php [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        $jobOffer = $this->factory->create(JobOffer::class, [
+            'title' => 'Site Reliability Engineering Manager',
+        ]);
+        $this->assertEquals("#{$jobOffer->getId()} Site Reliability Engineering Manager", (string) $jobOffer);
```
Comment (ianrodrigues) [inline tests/Unit/Entity/JobOfferTest.php]:
```suggestion

        $this->assertEquals("#{$jobOffer->getId()} Site Reliability Engineering Manager", (string) $jobOffer);
```

**Pull request merged**